src/test/java/org/jenkinsci/plugins/saml/SamlJCasCCompatibilityTest.java

```diff
@@ -0,0 +1,68 @@
+package org.jenkinsci.plugins.saml;
+
+import hudson.security.SecurityRealm;
+import io.jenkins.plugins.casc.misc.RoundTripAbstractTest;
+import org.jenkinsci.plugins.saml.conf.Attribute;
+import org.jenkinsci.plugins.saml.conf.AttributeEntry;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.hamcrest.CoreMatchers.containsString;
+
+public class SamlJCasCCompatibilityTest extends RoundTripAbstractTest {
+    @Override
+    protected void assertConfiguredAsExpected(RestartableJenkinsRule restartableJenkinsRule, String s) {
+        final SecurityRealm realm = restartableJenkinsRule.j.jenkins.getSecurityRealm();
+        assertTrue("The Security Realm should not be null", realm != null);
+        assertTrue("The Security Realm should be SamlSecurityRealm", realm instanceof SamlSecurityRealm);
+
+        final SamlSecurityRealm samlRealm = (SamlSecurityRealm)realm;
+        // Simple attributes
+        assertEquals(String.format("Wrong Display Name Attribute Name value. Expected %s but retrieved %s", "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name", samlRealm.getDisplayNameAttributeName()), "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name", samlRealm.getDisplayNameAttributeName());
```

Comment:
Isn't the difference already displayed by Assert?

Reply:
But only the difference, not the name of the field. Since there are so many fields to check, I think it's good to have that information in the message, so it's easier to match the the error with the field and the displayed text in the UI

Reply:
In my opinion if the test fails you are going to have to dig on it in both scenarios and finding out where the failure happens having the values should not be hard. I prefer the readability of code to be honest.